internal/corazawaf/transaction.go

```diff
@@ -1591,6 +1609,20 @@ func (tx *Transaction) generateResponseBodyError(err error) {
 	tx.variables.resBodyProcessorErrorMsg.Set(err.Error())
 }
 
+// setTimeVariables generates all the time variables
+func (tx *Transaction) setTimeVariables() {
+	timestamp := time.Unix(0, tx.Timestamp)
+	tx.variables.time.Set(timestamp.Format(time.TimeOnly))
+	tx.variables.timeDay.Set(strconv.Itoa(timestamp.Day()))
+	tx.variables.timeEpoch.Set(strconv.FormatInt(timestamp.Unix(), 10))
+	tx.variables.timeHour.Set(strconv.Itoa(timestamp.Hour()))
+	tx.variables.timeMin.Set(strconv.Itoa(timestamp.Minute()))
+	tx.variables.timeSec.Set(strconv.Itoa(timestamp.Second()))
+	tx.variables.timeWday.Set(strconv.Itoa(int(timestamp.Weekday())))
+	tx.variables.timeMon.Set(strconv.Itoa(int(timestamp.Month()) - 1))
```

Comment:
Having `TIME_MON` in the range `0-11` instead of `1-12` feels very counterintuitive to me.
ModSecurity behaviour is not consistent, with `v2` ranging `1-12` and `v3` ranging `0-11`. I opened this issue about it: https://github.com/owasp-modsecurity/ModSecurity/issues/3305.

I'm pushing for a unified behaviour between all the engines and I feel that the natural one would be `1-12` range.

Note: If we are going to implement it as `1-12` also Coraza Docs have to be updated

Reply:
Ok, I agree. Should I update this PR right now, or wait until Modsecurity issue is closed?

Reply:
It might take long

Reply:
So I can update in advance, right?

Reply:
We would not be compliant with one of the two current ModSecurity implementations anyway 😄. 
There is already an open PR https://github.com/owasp-modsecurity/ModSecurity/pull/3306 that aligns v3 to use `1-12` range. So I believe that we have pretty much agreed that `1-12` range is the one we should stick with. Let's go with it

Reply:
Fixed